dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/IdeController.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.dimensiondata.cloudcontrol.domain;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import java.util.List;
+
+@AutoValue
+public abstract class IdeController extends AbstractBaseController {
```

Comment:
Inheritance with Google auto value is not supported. You already have to define the parent class fields in the child class factory methods and builders, wo there is little gain abstracting the common accessors in the base class. There are just a few, so consider removing the base calss and putting all information here.

Also, inheritance is tricky: we annotate the "factory methods" with the json field names, not the individual fields. This is good for deserialization but when serializing we need to match each object field to its json name. Ideally we would annotate each field method, but that would mean duplicating the information from the factory method annotation. Instead, assuming the fact that Google Auto does not support inheritance, in jclouds we took the convention that object fields (well, abstract field accessors) would be decalred *in the same order* than the one of the factory method, and jclouds uses that convention when serializing. When there is a class hierarchy, field order is not that clear and more error-prone.

Reply:
Just seeing the classes below, you need to configure the order of the parameters in the factory methods matching the order in which the fields are declared. Double check that for all classes.